Movio/app/src/main/java/uco_396575/movio2/pv256/fi/muni/cz/movio/MainActivity.java

```diff
@@ -1,13 +1,79 @@
 package uco_396575.movio2.pv256.fi.muni.cz.movio;
 
-import android.support.v7.app.AppCompatActivity;
+import android.app.Fragment;
+import android.app.FragmentManager;
+import android.app.FragmentTransaction;
 import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
 
 public class MainActivity extends AppCompatActivity {
 
+    private final String DETAIL_TAG = "detail";
+    private final String DETAIL_FRAGMENT_TAG = "detailFragment";
+    private final String MAIN_FRAGMENT_TAG = "mainFragment";
+
+    private boolean isShowingDetail = false;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_main);
+        setContentView(R.layout.container_main);
+
+        FragmentManager fm = getFragmentManager();
+        FragmentTransaction ft = fm.beginTransaction();
+        ft.replace(R.id.list_fragment, MainFragment.newInstance(), MAIN_FRAGMENT_TAG).commit();
+    }
+
+    @Override
+    public void onBackPressed() {
+        FragmentManager fm = getFragmentManager();
+        fm.popBackStack();
+        isShowingDetail = false;
+    }
+
+    @Override
+    protected void onRestoreInstanceState(Bundle savedInstanceState) {
+        isShowingDetail = savedInstanceState.getBoolean(DETAIL_TAG);
+        removeDetailFragment();
+        if (isShowingDetail) addDetailFragment();
```

Comment:
je lepsie vzdy pouzivat zatvorky pre if. pomaha to zabranit nechcenym chybam v buducnosti